src/TMDBApp.Core/Views/PrincipalPage.xaml.cs

```diff
@@ -0,0 +1,23 @@
+﻿
+using TMDBApp.Core.ViewModels;
+
+namespace TMDBApp.Core.Views
+{
+	public partial class PrincipalPage
+    {
+	    protected MovieListViewModel MovieListViewModel => MovieListView.BindingContext as MovieListViewModel;
+
+
+        public PrincipalPage()
+		{
+			InitializeComponent();
+		}
+
+	    protected async override void OnAppearing()
+	    {
+	        base.OnAppearing();
+
+	        await MovieListViewModel.InitializeAsync(null);
```

Comment:
Could have used NavigationService to create the MainPage as well to use the service to call `InitializeAsync()`.